### ISSUE TITLE
Does not serialize on setItem

### ISSUE THREAD

**Issue by DanLatimer** [NathanaelA/nativescript-localstorage#17]
This module does not act the same as localstorage in a browser. In a browser localStorage.setItem will immediately serialize the object to a string. This results in objects being turned into `"[object Object]"` however in this plugin it simply saves whatever is passed to it as an object. This leads to inconsistent behaviour that can cause difficult debugging. The code will work until the application is closed and the plugin attempts to retrieve the objects from their serialized form which will produce the "[object Object]" version.

I realize that you shouldn't be attempting to store objects but it's lead to some confusion with some of our junior team members and it would be better if this plugin's getItem resulted in a consistent result.

Chrome:
```
localStorage.setItem('key', {a: 'hi', b:123})
undefined
localStorage.getItem('key')
"[object Object]"
```

nativescript-localstorage:
```
localStorage.setItem('key', {a: 'hi', b:123})
undefined
localStorage.getItem('key')
{a: 'hi', b:123}
// after the app is closed and reopened
localStorage.getItem('key')
"[object Object]"
```

**Comment by NathanaelA** [comment]
Yeah, that is an interesting side-effect.   I'll have to think about this; your point does make some sense...

**Comment by NathanaelA** [comment]
Hmm, after doing some tests; I'm getting a different result; is it possible something is doing a json.parse after you get the value back?:
```
localStorage.setItem('key', {a: 'hi', b:123})
localStorage.getItem('key')
{a: 'hi', b:123}
// after the app is closed and reopened
localStorage.getItem('key')
{a: 'hi', b:123}
```
However, this does not actually follow the spec; the spec says strings only.    So code written for standard localstorage would probably break if it had auto-object handling in it...

So this should be fixed in the next version, I have added code to auto force the value to be a string; so it should automatically get ["object Object"] when you try and do a `setItem("key", {a:1, b: "Blah"});`

**Comment by NathanaelA** [comment]
This is fixed in v2.0, since it is a breaking change.

**Comment by DanLatimer** [comment]
Thanks @NathanaelA that sounds like a good change
